csrc/loader.cpp

```diff
@@ -65,20 +70,28 @@ JNIEXPORT jstring JNICALL Java_com_amazon_corretto_crypto_provider_Loader_getNat
 
 }
 
-JNIEXPORT jboolean JNICALL Java_com_amazon_corretto_crypto_provider_Loader_validateLibcryptoExactVersionMatch(JNIEnv* pEnv, jclass)
-{
-    char msg_buffer[256] = {0};
+void accpValidateLibcryptoVersion(bool fuzzyMatch) {
+    unsigned long libcrypto_compiletime_version = OPENSSL_VERSION_NUMBER;
+    unsigned long libcrypto_runtime_version = OpenSSL_version_num();
 
-    try {
-        const unsigned long libcrypto_compiletime_version = OPENSSL_VERSION_NUMBER;
-        const unsigned long libcrypto_runtime_version = OpenSSL_version_num();
+    if (fuzzyMatch) {
+        libcrypto_compiletime_version &= LIBCRYPTO_MAJOR_MINOR_VERSION_MASK;
+        libcrypto_runtime_version     &= LIBCRYPTO_MAJOR_MINOR_VERSION_MASK;
+    }
 
-        if (libcrypto_compiletime_version != libcrypto_runtime_version) {
-            snprintf(msg_buffer, sizeof(msg_buffer), "Runtime libcrypto version does not match compile-time version. "
-                "Expected: 0x%08lX , Actual: 0x%08lX", libcrypto_compiletime_version, libcrypto_runtime_version);
-            throw java_ex(EX_RUNTIME_CRYPTO, msg_buffer);
-        }
+    if (libcrypto_compiletime_version != libcrypto_runtime_version) {
+        memset(accp_loader_exception_msg, 0, sizeof(accp_loader_exception_msg));
+        snprintf(accp_loader_exception_msg, sizeof(accp_loader_exception_msg),
+                "Runtime libcrypto version does not match compile-time version. Expected: 0x%08lX , Actual: 0x%08lX",
+                libcrypto_compiletime_version, libcrypto_runtime_version);
+        throw java_ex(EX_RUNTIME_CRYPTO, accp_loader_exception_msg);
+    }
+}
 
+JNIEXPORT jboolean JNICALL Java_com_amazon_corretto_crypto_provider_Loader_validateLibcryptoExactVersionMatch(JNIEnv* pEnv, jclass)
```

Comment:
NitPick: I'd just expose a single native method and take in a boolean as a parameter.